src/events/events-router.js

```diff
@@ -4,19 +4,143 @@ const eventsRouter = express.Router();
 const jsonParser = express.json();
 const xss = require('xss');
 
-// HELPER:
+const expectedKeys = ['title', 'categories', 'description'
+                      , 'event_img', 'source_name', 'source_url'
+                      , 'source_img', 'date_published'];
+
+// validate that event obj has all required fields
+const validateKeys = (event) => {
+  let missingKeys = [];
+  expectedKeys.forEach(key => {
+    if(!(event[key]))
+      missingKeys.push(key);
+  });
+  return missingKeys.join(', ');
+}
+
+// serialize and sanitize event
 const serializeEvent = (event) => {
-  /* IMPLEMENT ME */
+
+  try {
+    if (!event || !Object.keys(event).length)
+      throw new Error('A not empty comment object must be provided');
+    
+    const missingKeys = validateKeys(event);
+    if(missingKeys !== '')
+      throw new Error(`Fields ${missingKeys} must be provided`);
+    return {
+      title: xss(event.title),
+      categories: xss(event.categories),
+      description: xss(event.description),
+      event_img: xss(event.event_img),
+      source_name: xss(event.source_name),
+      source_url: xss(event.source_url),
+      source_img: xss(event.source_img),
+      date_published: new Date(xss(event.date_published))
+    };
+  }
+  catch (er) {
+    console.log('error: ', er);
+    return { error: er}
+  }
 };
 
+const serializeCategory = (category) => {
+  return xss(category)
+}
+
 // TODO /events/
 eventsRouter
   .route('/')
-  .get((req, res, next) => {
-    /* IMPLEMENT ME */
+  .get(jsonParser, (req, res, next) => {
+    const { category } = req.query;
+    console.log('incoming category requested is: ', category);
+    const sanitizedCategory = serializeCategory(category);
+    console.log('sanitizedCategory is: ', sanitizedCategory);
+    if(!sanitizedCategory || sanitizedCategory === 'US'){
+      EventsService.getEvents(req.app.get('db'))
+        .then( events => {
+          res.status(200).json({ events })
+        })
+        .catch( er => {
+          console.log('er at 66', er);
+          next(er);
+        })
+    }
+    else {
+      console.log('about to search for a specific category');
+      EventsService.getEventsByCategory(req.app.get('db'), sanitizedCategory)
+        .then( events => {
+          res.status(200).json({ events })
+        })
+        .catch( er => {
+          console.log('er at 76', er);
+          next(er);
+        })
+    }
   })
   .post(jsonParser, (req, res, next) => {
```

Comment:
post to /events should have some kind of auth. i implemented bearer token auth with a secret worker key, but we can use something else. In the end, it turns out CORS only works for browser execution; you can override the host header on the request outside of the browser, which makes the endpoint less secure than bearer auth. I have a middleware function (requireWorkerAuth) that authenticates the worker. Lmk what you think

Reply:
Also lmk if you want me to use .then().catch() notation over try/catch